src/Core/src/Platform/Android/Navigation/NavigationRootManager.cs

```diff
@@ -134,6 +134,8 @@ void SetContentView(IView? view)
 				}
 
 				_viewFragment = null;
+
+				RootViewChanged?.Invoke(this, EventArgs.Empty);
```

Comment:
Should be fixed with https://github.com/dotnet/maui/pull/20052/commits/2f4cd0cfafc2954828282f06002e6b8aaa6a07d6, as I wrote below it's unrelated to what I was addressing with this PR (you would have hit this in current MAUI stable switching any Window main page.)

The issue is the way the layers work with Android and where the graphics layout is put. When we initialize the VisualDiagnosticsOverlay for Android, it goes on the top most layout of the view, which just so happens to be whatever the current Page content for a given Window is. That is unlike iOS/Mac/Windows where it's put above every MAUI view so it works regardless of switching out the page. With Android we need to recreate the overlay if you swap out the MainPage of a given window, since swapping it takes out the graphics layer.